File: src/main/java/net/golem/raknet/protocol/unconnected/UnconnectedPingPacket.java
package net.golem.raknet.protocol.unconnected;

import net.golem.raknet.codec.PacketDecoder;
import net.golem.raknet.codec.PacketEncoder;
import net.golem.raknet.protocol.DataPacket;
import net.golem.raknet.protocol.RakNetPacketIds;

public class UnconnectedPingPacket extends DataPacket {

	public long pingId;

	public long clientGuid;

	public UnconnectedPingPacket() {
		super(RakNetPacketIds.UNCONNECTED_PING);
	}

	@Override
	public void encode(PacketEncoder encoder) {
		encoder.writeLong(pingId);
		encoder.writeMagic();
		encoder.writeLong(clientGuid);
	}

	@Override
	public void decode(PacketDecoder decoder) {
		pingId = decoder.readLong();
		decoder.skipMagic();
		clientGuid = decoder.readLong();
	}

	@Override
	public String toString() {
		return "UnconnectedPingPacket{" +
				"pingId=" + pingId +
				", clientGuid=" + clientGuid +
				'}';
	}
}

File: src/main/kotlin/raknet/packet/protocol/UnconnectedPingPacket.kt
package raknet.packet.protocol

import io.netty.buffer.ByteBuf
import raknet.Magic
import raknet.packet.DataPacket
import raknet.packet.PacketType
import raknet.packet.readMagic

class UnconnectedPingPacket(
    var time: Long,
    var magic: Magic = Magic,
    var clientGuid: Long
): DataPacket(PacketType.UNCONNECTED_PING.id()) {

    override fun decode(buffer: ByteBuf) {

    }

    companion object {
        fun from(buffer: ByteBuf): UnconnectedPingPacket {
            return UnconnectedPingPacket(
                buffer.readLong(),
                buffer.readMagic(),
                buffer.readLong()
            )
        }
    }

    override fun toString(): String {
        return "UnconnectedPingPacket(time=$time, clientGuid=$clientGuid)"
    }
}